integration-tests/test/shared/docker-compose.ts

```diff
@@ -0,0 +1,29 @@
+import * as compose from 'docker-compose'
+import * as shell from 'shelljs'
+import * as path from 'path'
+
+
+type ServiceNames = 'batch_submitter' | 'dtl' | 'l2geth' | 'relayer'
+
+const OPS_DIRECTORY = path.join(process.cwd(), '../ops')
+const DEFAULT_SERVICES: ServiceNames[] = ['batch_submitter', 'dtl', 'l2geth', 'relayer']
+
+export class DockerComposeNetwork {
+    constructor(private readonly services: ServiceNames[] = DEFAULT_SERVICES) {}
+
+    async up() {
+        const out = await compose.upMany(this.services, {cwd: OPS_DIRECTORY})
+
+        const {err, exitCode} = out;
+
+        if (!err || exitCode) {
+            console.error(err)
+            throw new Error('Unexpected error when starting docker-compose network, dumping output')
+        }
+
+        if (err.includes('Creating')) {
+            console.info('🐳 Tests required starting containers. Waiting for sequencer to ready.')
+            shell.exec(`${OPS_DIRECTORY}/scripts/wait-for-sequencer.sh`, {cwd: OPS_DIRECTORY})
+        }
+    }
```

Comment:
What do you think about adding a `down` method and then overwriting the hardhat test task such that it calls `up` once and then runs all the tests with `super` and then calls `down` once?

Or just adding a call to down in a script that gets called after the npm script? Something like:
```
"test:integration":  "hardhat test && node scripts/dc-down.js"
```

Reply:
If we want to do that, what about using [Mocha global setup & teardown instead](https://futurestud.io/tutorials/mocha-global-setup-and-teardown-before-after)?

Reply:
I didnt know there was a way to use mocha globally like that - it was my first thought so i'd be down if that is your preference